cobc/flag.def

```diff
@@ -282,3 +282,12 @@ CB_FLAG (cb_diagnostics_absolute_paths, 1, "diagnostics-absolute-paths",
 CB_FLAG (cb_flag_prof, 1, "prof",
 	_("  -fprof                enable profiling of the COBOL program"))
 
+CB_FLAG_NQ (1, "dump-tree", CB_FLAG_DUMP_TREE,
+	_("  -fdump-tree=<file>    dump parsed AST to <file>, format is JSON unless\n"
+	  "                        the file has a .ml extension for OCaml format."))
+
+CB_FLAG_NQ (1, "dump-tree-flags", CB_FLAG_DUMP_TREE_FLAGS,
+	_("  -fdump-tree-flags=<flags>    set flags used when dumping the parsed AST\n"
+	  "                        '+/-' to enable/disable, 'l' for locations,\n"
+	  "                        't' for types, 'p' for pointers, 'i' for indent."))
```

Comment:
this misses some of the flags

Reply:
Do you consider the "no JSON/OCaml" temporary or is the idea to leave that as-is? Just asking....

Reply:
Initially, I thought that the AST of GnuCOBOL was "clean", in the sense that it was an exact representation of the COBOL code. However, it is not the case: the parser starts translating code to C (for example, replacing some nodes by direct calls to C functions), so the AST is actually half COBOL code, half translated code. So, the interest of sending this AST to another program (through OCaml or JSON) is less interesting. So, for now, I have given up on the idea of a correct JSON/OCaml encoding, focusing more on the ability to debug the compiler using a printed AST.

Yet, on the long term, I think it would be nice to move translations done in the parser to the codegen, and get a clean AST. At that point, it could be used by other programs and this output could be improved.

Reply:
I totally agree that this separation should be clearer; to do so we'd likely move most of the included C code to a new cb_tree like it currently is implemented, but into a new type of node that is only converted to C in codegen.
... but then - nearly all of the C parts should be FUNC_CALLs and most of those should be quite easily be separated and could be converted "on the fly", if wanted.

... but your human-readable AST dump may be ideal to show what is really in and consider adjustments that way.
I'm looking forward to some test cases showing what you consider to be bad in the AST. :-)

Reply:
Here is an example of the generated output for a small file:

[writefile.txt](https://github.com/user-attachments/files/17522224/writefile.txt)

I am not really sure it is interesting to create a test, as we don't really expect the format to be kept backward compatible. We may just want to make a test to verify that a file is generated, but not check the file format itself.